pkg/ruler/notifier.go

```diff
@@ -16,8 +17,21 @@ import (
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/dns"
 	"github.com/prometheus/prometheus/notifier"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/tls"
 )
 
+type NotifierConfig struct {
+	TLS       tls.ClientConfig `yaml:"tls"`
+	BasicAuth util.BasicAuth   `yaml:"basic_auth"`
+}
+
+func (cfg *NotifierConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.TLS.RegisterFlagsWithPrefix("ruler.alertmanager.client", f)
+	cfg.BasicAuth.RegisterFlagsWithPrefix("ruler.alertmanager.client.", f)
```

Comment:
The yaml option is `alertmanager_client`, so the flags prefix here should be `ruler.alertmanager-client.`

Reply:
Also, the basic auth flags prefix should include it, so `ruler.alertmanager-client.basic-auth.`

Reply:
👍